roles/edpm_kernel/tasks/kernelargs.yml

```diff
@@ -143,6 +143,18 @@
       ansible.builtin.set_fact:
         reboot_required: true
 
+    - name: Create a reboot_required directory if it does not exist
+      ansible.builtin.file:
+        path: /var/lib/openstack/reboot_required/
+        state: directory
+        mode: '0755'
+    
+    - name: Create file for reboot required
+      ansible.builtin.file:
+        path: /var/lib/openstack/reboot_required/needs_restarting
+        state: touch
+        mode: '0600'
```

Comment:
this is already happening when reboot.yaml is included. I'd check the ansible log and make sure that was included, if it wasn't, maybe it's the conditions on that task.

Reply:
Yeah, I think because we don't call the `reboot.yml` if the `grub.cfg` is changed:
https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_kernel/tasks/kernelargs.yml#L161-L164

I think we probably DO want to call that `reboot.yml` in this case. Maybe just removing the `NOT` from the condition would fix the issue I'm seeing with nodes not being rebooted when kernel args are updated.

Reply:
Converting this to draft while I'm testing it, but I'll try with:
https://github.com/openstack-k8s-operators/edpm-ansible/pull/648/files#diff-42b415cea97608f4334dee2bbaf5a43418ae1318c0e1a926b8aa459354a0ca1aR158